generate.sh

```diff
@@ -5,7 +5,28 @@ if [ $# -eq 0 ]; then
     exit 1
 fi
 
-if command -v podman > /dev/null
+
+get_container_engine () {
+
+    # allow override from callers such as oci-env
+    if [[ -z ${COMPOSE_BINARY} ]]; then
```

Comment:
You missed by `-n` vs `-z` comment.
```suggestion
    if [[ -n ${COMPOSE_BINARY} ]]; then
```